src/Arcus.Security.Core/CompositeSecretProvider.cs

```diff
@@ -100,25 +107,30 @@ internal class CompositeSecretProvider : ICachedSecretProvider, ISecretStore
             {
                 return concreteProvider;
             }
+            
+            if (provider is CompositeSecretProvider)
+            {
+                throw new InvalidOperationException(
+                    $"Cannot cast registered {nameof(ISecretProvider)} with name '{name}' to type '{typeof(TSecretProvider).Name}' " +
+                    $"because multiple secret providers were registered with the name '{name}'");
```

Comment:
Maybe an idea to include in the exception message how this can be solved ?  Maybe mention that you should use the non generic method to retrieve the secretproviders if multiple secretproviders have been registered with the same name ?

For instance:
>"Multiple SecretProviders have been registered with name {name}.  Use the non-generic GetProvider method to retrieve them"

Reply:
Yes, thanks, did that. Resolving.